libr/util/calc.c

```diff
@@ -227,7 +227,7 @@ static int cin_get_num(RNum *num, RNumCalc *nc, RNumCalcValue *n) {
 			str[i++] = c;
 		}
 	}
-	str[i] = 0;
+	str[i-1] = 0;
```

Comment:
can't we allocate at https://github.com/radare/radare2/pull/11498/files#diff-8fd9931ac570e117a8fe75fcdb73389aR217 a string of R_NUMCALC_STRSZ + 1? I'm not sure the logic is still the same with this `i-1`.

Reply:
So I think I wasn't sure yesterday, and I guess both would work.
However, I assumed that this would be consistent of others R_NUMCALC_STRSZ use, such as https://github.com/radare/radare2/blob/master/libr/util/calc.c#L341  The name also imply that the size count the final 0.